python/ray/tests/test_advanced_3.py

```diff
@@ -785,6 +786,20 @@ def nested_get(self, key):
         }).remote("z")) == "job_z")
 
 
+def test_get_conda_env_dir(tmp_path):
+    d = tmp_path / "tf1"
+    d.mkdir()
+    os.environ["CONDA_PREFIX"] = str(d)
+    with pytest.raises(ValueError):
+        # Env tf2 should not exist.
+        env_dir = get_conda_env_dir("tf2")
+    tf2_dir = tmp_path / "tf2"
+    tf2_dir.mkdir()
+    env_dir = get_conda_env_dir("tf2")
+    assert (env_dir == str(tmp_path / "tf2"))
+    os.environ["CONDA_PREFIX"] = ""
+
+
```

Comment:
is it also possible to have an end-to-end test that creates two different conda envs and starts actors in each? we should also test the inheritance properties (child tasks/actors will be in same env)

Reply:
We wondered about this back when we did the Serve conda feature, but I don't think we were able to figure out how to do it--I'll continue to think about it.  For the underlying environment variable override feature, the inheritance properties are tested, but you're right that we should still have an end-to-end test for the conda env feature.